onos_ric_sdk_py/sdl.py

```diff
@@ -74,6 +75,96 @@ async def get_cell_ids(self, e2_node_id: str) -> List[str]:
         except GRPCError as e:
             raise ClientRuntimeError() from e
 
+    async def _get_cell_entity_id(self, e2_node_id: str, cell_id: str) -> int:
+        """
+        given e2_node_id and cell_id, returns entity id
+        raises ClientRuntimeError if cell_id is not found
+        """
+        if not self._ready:
+            raise ClientStoppedError()
+
+        client = TopoStub(self._topo_channel)
+        filters = Filters(
+            relation_filter=RelationFilter(
+                src_id=e2_node_id,
+                relation_kind=RanRelationKinds.CONTAINS.name.lower(),
+                target_kind="",
+            )
+        )
+
+        try:
+            response = await client.list(filters=filters)
+            for obj in response.objects:
+                if obj.entity.kind_id != RanEntityKinds.E2CELL.name.lower():
+                    continue
+
+                aspects = obj.aspects["onos.topo.E2Cell"].value
+                e2_cell = E2Cell().from_json(aspects)
+                if e2_cell.cell_object_id == cell_id:
+                    return obj.id
+        except GRPCError as e:
+            raise ClientRuntimeError() from e
+
+        raise ClientRuntimeError(
+            f"cannot find cell_id:{cell_id} in e2_node:{e2_node_id}"
+        )
+
+    async def get_cell_data(
+        self, e2_node_id: str, cell_id: str, key: str
+    ) -> Optional[bytes]:
+        """
+        get data referenced by key attached to a cell_id
+        """
+
+        if not self._ready:
+            raise ClientStoppedError()
+
+        entity_id = await self._get_cell_entity_id(e2_node_id, cell_id)
+
+        client = TopoStub(self._topo_channel)
+        try:
+            resp = await client.get(id=entity_id)
+        except GRPCError as e:
+            raise ClientRuntimeError() from e
+
+        type_data = resp.object.aspects.get(key)
+        if type_data is None:
+            return None
+
+        return type_data.value
+
+    async def set_cell_data(
+        self, e2_node_id: str, cell_id: str, key: str, data: bytes
+    ) -> None:
+        """
+        set data referenced by key attached to a cell_id
+        remove data referenced by key if data is None
+        """
+
+        if not self._ready:
+            raise ClientStoppedError()
+
+        entity_id = await self._get_cell_entity_id(e2_node_id, cell_id)
+
+        client = TopoStub(self._topo_channel)
+        try:
+            resp = await client.get(id=entity_id)
+        except GRPCError as e:
+            raise ClientRuntimeError() from e
+
+        if data is None:
+            if key not in resp.object.aspects:
+                return
+            else:
+                del resp.object.aspects[key]
+        else:
+            resp.object.aspects[key] = betterproto.lib.google.protobuf.Any(key, data)
```

Comment:
Does the value need to be a `betterproto` `Any` object?

Reply:
the values are defined in the proto file as google.protobuf.Any

https://github.com/onosproject/onos-api/blob/master/proto/onos/topo/topo.proto#L174